File: web/tests/reference/graphs/course-schedule-ii.js
// Drains the frontier as a STACK where the Python reference uses a queue, so
// the two produce different valid orders on the same input — the reason this
// lesson is graded by a property.
function findOrder(numCourses, prerequisites) {
  const adj = Array.from({ length: numCourses }, () => []);
  const indeg = new Array(numCourses).fill(0);
  for (const [a, b] of prerequisites) {
    adj[b].push(a);
    indeg[a] += 1;
  }
  const stack = [];
  for (let i = 0; i < numCourses; i++) if (indeg[i] === 0) stack.push(i);
  const order = [];
  while (stack.length) {
    const node = stack.pop();
    order.push(node);
    for (const next of adj[node]) if (--indeg[next] === 0) stack.push(next);
  }
  return order.length === numCourses ? order : [];
}

File: web/tests/reference/graphs/course-schedule-ii.py
from collections import deque

def find_order(num_courses, prerequisites):
    adj = [[] for _ in range(num_courses)]
    indeg = [0] * num_courses
    for a, b in prerequisites:
        adj[b].append(a)
        indeg[a] += 1
    queue = deque(i for i in range(num_courses) if indeg[i] == 0)
    order = []
    while queue:
        node = queue.popleft()
        order.append(node)
        for nxt in adj[node]:
            indeg[nxt] -= 1
            if indeg[nxt] == 0:
                queue.append(nxt)
    return order if len(order) == num_courses else []
